pkg/controller/builder.go

```diff
@@ -185,3 +195,60 @@ func replaceImageHook(placeholder string, desiredImageEnvVar string) deploymentc
 		return []byte(newDeployment), nil
 	}
 }
+
+func updateEnv(con *corev1.Container, name, value string) ([]corev1.EnvVar, error) {
+	for _, e := range con.Env {
+		if e.Name == name {
+			return con.Env, fmt.Errorf("unexpected environment variable %q=%q in container %q while building manifests", e.Name, e.Value, con.Name)
+		}
+	}
+	if value == "" {
+		return con.Env, nil
+	}
+	klog.FromContext(context.Background()).WithName("builder").V(1).Info("Updated environment", "container", con.Name, "key", name, "value", value)
+	return append(con.Env, corev1.EnvVar{Name: name, Value: value}), nil
+}
+
+func setProxyEnvs(con *corev1.Container, vars map[string]string) error {
+	var err error
+	// The order these are put into the Container matter for hashing
+	// The ordering of maps varies by design in golang, so we have to sort a slice of keys
+	keys := maps.Keys(vars)
+	slices.Sort(keys)
+	for _, key := range keys {
+		con.Env, err = updateEnv(con, key, vars[key])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontroller.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		proxyConfig, err := pc.Get("cluster")
```

Comment:
@tmshort I guess the proxy informer would provide us with the cluster wide proxy info. Is that right?

Reply:
Yes, and it's called `cluster`